crates/bevy_render/src/world_sync.rs

```diff
@@ -0,0 +1,126 @@
+use std::{marker::PhantomData, ops::DerefMut};
+
+use bevy_app::Plugin;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    bundle::Bundle,
+    component::Component,
+    entity::{Entity, EntityHashMap},
+    observer::Trigger,
+    query::With,
+    reflect::ReflectComponent,
+    system::{Query, ResMut, Resource},
+    world::{Mut, OnAdd, OnRemove, World},
+};
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_reflect::Reflect;
+
+/// Marker component that indicate its entity needs to be Synchronized to render world
+#[derive(Component, Clone, Debug, Default, Reflect)]
+#[reflect[Component]]
+pub struct ToRenderWorld;
```

Comment:
Is adding this the user's responsibility? That seems like it could be a significant footgun, but maybe I've just missed where/if it's automatically added. I'd love if this were just part of ExtractComponentPlugin. And actually, does this have any use outside of being a catch-all to trigger WorldSyncPlugin's observers?

Reply:
No, this is just a manual tagging, and in most cases it is not necessary to use it. If users need to synchronize the entity of a specific component, they should directly use WorldSsyncPlugin<Component>,but it can also be useful in certain situations(`SpatialBundle`)